src/main/python/esdt/pmf.py

```diff
@@ -1,69 +1,175 @@
+        d = self.extract_data()
+        ll = np.zeros(params.shape[1], 'd')
+        for i in range(params.shape[1]):
+            ll[i] = self.negloglikelihood(params[:, i], d.x, d.k, d.n)
+        for par, prior in zip(params, self.priors):
+            ll += prior.logpmf(par)
+        ll -= ll.max()  # To avoid overflow, posterior is unnormalized anyways
+        np.exp(ll, ll)
+        return ll
+
+    def extract_data(self):
+        Dataset = namedtuple('Dataset', ['x', 'k', 'n', 'y'])
+        return Dataset(self.data[:, 0],
+                       self.data[:, 1]*self.data[:, 2],
+                       self.data[:, 2],
+                       self.data[:, 1])
+
+    def jackknife_sem(self, data):
+        params = []
+        for i in range(data.shape[0]):
+            mask = np.ones(data.shape[0], bool)
+            mask[i] = False
+            params.append(self.fit(data[mask],
+                                   assign=False,
+                                   start=self.params)[0])
+        v = (len(params)-1)*np.var(np.array(params), 0)
+        r = np.corrcoef(np.array(params).T)
+        se = np.sqrt(v/len(params))
+        self.sem = se
+        self.param_r = r
+        return se, r
+
+    def predict(self, x, params=None):
+        if params is None:
+            params = self.params
+        if self.guess is None:
+            guess = params[0]
+            params = params[1:]
+        else:
+            guess = self.guess
+        return guess + (1-guess-params[-1])*self.F(x, params[0], params[1])
+
+    def get_start(self, x, y):
+        if self.guess is None:
+            guess = 0.02
+        else:
+            guess = self.guess
+        y_ = (y-guess)/(1-guess)
+        y = y_
+        i = np.logical_and(y > 0, y < 1)
+        y = y[i]
+        logit = np.log(y/(1-y))
+        slope, intercept = stats.linregress(x[i], logit)[:2]
+        th = -intercept/slope
+        w = 2/slope
+        if self.guess is None:
+            return np.array([th, w, 0.02, 0.02])
+        else:
+            return np.array([th, w, 0.02])
+
+
+def bayesian_inference(pmf, statistics={'threshold': lambda grid: grid[0],
+                                        'width': lambda grid: grid[1]}):
+    grid = mkgrid(
+        *[(par - 3*se, par + 3*se, 20) for par, se in zip(pmf.params, pmf.sem)]
+    )
+    grid, posterior = integrate_posterior(pmf, grid)
+    return {key: get_stats(posterior, getter(grid))
+            for key, getter in statistics.items()}
+
+
+def pmfplot(pmf, **kwargs):
+    if 'axes' in kwargs:
+        ax = kwargs.pop('axes')
+    else:
+        if HASPYLAB:
+            ax = pl.axes()
+        else:
+            raise RuntimeError(
+                'Trying to use pylab function by pylab does not exist')
+    col = kwargs.setdefault('color', 'black')
+
+    dataset = pmf.extract_data()
+    xmin = dataset.x.min()
+    xmax = dataset.x.max()
+    x = kwargs.pop('x', np.mgrid[xmin:xmax:100j])
+    i0 = x <= xmin
```

Comment:
I seem to have changed the color of this line to yellow, i'm not sure what that means but there is nothing wrong with this line.